listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -360,3 +371,44 @@ def get_recording_recommendation_events(users_for_events: List[dict], min_ts: in
             current_app.logger.error('Validation error: ' + str(e), exc_info=True)
             continue
     return events
+
+
+def get_recording_pin_events(users_for_events: List[dict], min_ts: int, max_ts: int, count: int) -> List[APITimelineEvent]:
+    """ Gets all recording pin events in the feed."""
+
+    id_username_map = {user['id']: user['musicbrainz_id'] for user in users_for_events}
+    recording_pin_events_db = get_pins_for_feed(
+        user_ids=(user['id'] for user in users_for_events),
+        min_ts=min_ts,
+        max_ts=max_ts,
+        count=count,
+    )
+
+    events = []
+    for pin in recording_pin_events_db:
+        try:
+            pin = fetch_track_metadata_for_pin(pin)
```

Comment:
This is a bit of a problem here -- given that the fetch_track_metadata_for_pin is in a loop, it means that we're going to do 1 DB query for each pin event, which is far from ideal. We need to support doing this in one query -- one possibility is to refactor the fetch_track_metadata_for_pin to accept a list of pins, rather than just one.

Reply:
Thanks, I opened #1570 to refactor the functions and fix this, after that gets merged we should be able to merge this PR.

Reply:
OK, 1570 is merged.

Reply:
Fix the merge conflict on 1560 and lets get this merged!